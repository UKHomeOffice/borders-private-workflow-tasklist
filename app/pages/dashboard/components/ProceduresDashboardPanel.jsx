import React from "react";
import {withRouter} from "react-router";

class ProceduresDashboardPanel extends React.Component {


    procedures(e) {
        e.preventDefault();
        this.props.history.replace({
            pathname: "/procedures", state: {
                shiftPresent: this.props.hasActiveShift
            }
        });
    }

    render() {
        return <li className="__card column-one-third" id="proceduresPanel">
            <a href="#" onClick={this.procedures.bind(this)} className="card__body">
                <span className="bold-large">procedures</span>
            </a>
            <div className="card__footer">
                <span className="font-small">Operational procedures</span>
            </div>
        </li>
    }
}

export default withRouter(ProceduresDashboardPanel)