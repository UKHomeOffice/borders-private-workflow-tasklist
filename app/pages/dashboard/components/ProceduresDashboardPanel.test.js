import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ProceduresDashboardPanel } from './ProceduresDashboardPanel';

describe('Procedures Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'procedures-page': {},
  };
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it('renders procedures dashboard panel', async () => {
    const props = {};
    const wrapper = await mount(<ProceduresDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('navigates to procedures page on click', async () => {
    const history = createMemoryHistory('/procedures');

    const props = {
      history,
      hasActiveShift: true,
    };
    const wrapper = await mount(<Router history={history}>
      <ProceduresDashboardPanel
        store={store}
        {...props}
      />
                                </Router>);

    const proceduresPageLink = wrapper.find('#proceduresPageLink');
    expect(proceduresPageLink.exists()).toEqual(true);

    proceduresPageLink.simulate('click');
    expect(props.history.location.pathname).toEqual('/procedures');
  });
});
