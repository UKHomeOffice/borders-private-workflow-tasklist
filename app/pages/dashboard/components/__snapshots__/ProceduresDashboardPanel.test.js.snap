// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procedures Dashboard Panel renders procedures dashboard panel 1`] = `
<ProceduresDashboardPanel
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card govuk-grid-column-one-third"
    id="proceduresPanel"
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    <a
      className="card__body"
      href="#"
      id="proceduresPageLink"
      onClick={[Function]}
    >
      <span
        className="govuk-!-font-size-36 govuk-!-font-weight-bold"
      >
        Procedures
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="govuk-!-font-size-19"
      >
        Start a procedure
      </span>
    </div>
  </li>
</ProceduresDashboardPanel>
`;
